tools/crm_diff.c

```diff
@@ -209,13 +209,20 @@ generate_patch(xmlNode *object_1, xmlNode *object_2, const char *xml_file_2,
                gboolean as_cib, gboolean no_version)
 {
     xmlNode *output = NULL;
+    int rc = pcmk_rc_ok;
+
+    pcmk__output_t *out = NULL;
+    int out_rc = pcmk_rc_no_output;
 
     const char *vfields[] = {
         XML_ATTR_GENERATION_ADMIN,
         XML_ATTR_GENERATION,
         XML_ATTR_NUMUPDATES,
     };
 
+    rc = pcmk__log_output_new(&out);
```

Comment:
Is this going to conflict when you add formatted output to crm_diff? The code here will always want a logger, whereas the overall output object could have any format

Reply:
It probably will. I was thinking the non-logger `out` object could be local to `main()`, but a handful of helper functions use it, so it'll be better to make it global.

I could change the name of this variable later when we get to that point, but since I have a couple more changes to make prior to merge, I might as well do it now.